apps/minaintyg/src/pages/Certificates/components/CertificateList.tsx

```diff
@@ -0,0 +1,33 @@
+import { getYear, parseISO } from 'date-fns'
+import { CertificateListItem } from '../../../schema/certificateList.schema'
+import { CertificateCard } from './CertificateCard/CertificateCard'
+import { SortingOrder } from './CertificateListOrder'
+
+const getCertificateYear = ({ timestamp }: CertificateListItem) => getYear(parseISO(timestamp))
+
+export function CertificateList({ certificates, order = 'descending' }: { certificates: CertificateListItem[]; order?: SortingOrder }) {
+  const years = certificates.reduce((result, certificate) => result.add(getCertificateYear(certificate)), new Set<number>())
+  return (
+    <div>
+      {Array.from(years)
+        .sort((a, b) => (order === 'ascending' ? a - b : b - a))
+        .map((year) => (
+          <div key={year} className="[&:not(:last-child)]:mb-10" data-testid={`${year}-certificates`}>
+            <h2 className="ids-heading-2 mb-5" key={year}>
+              {year}
+            </h2>
+            {[...certificates]
+              .filter((certificate) => getCertificateYear(certificate) === year)
+              .sort((a, b) =>
```

Comment:
Can this be extracted into a function to add readability?

Reply:
Good call, moved business logic to its own file.